### PR TITLE
Variable port. Restart geth if out of sync [ch359]

### PR DESCRIPTION
**Pull request by kivutar**

### CODE REVIEW

File: outOfSyncNoti.sh [hunk 1]
```diff
@@ -1,14 +1,22 @@
 #!/bin/bash
-# A simple shell script to check if node is out of sync with ethereum testnet
-# If node is out of sync, user will be nptofied by OpsGenie
-# Using crontab command: `*/10 * * * * /home/someuser/tmp/` to run this script every 10 mins.
+# A simple shell script to check if node is out of sync with ethereum
+# If node is out of sync, user will be notified by OpsGenie
+# Using crontab command: `*/10 * * * * /home/foo/outOfSyncNoti.sh 'rinkeby' 'opsgeniekeyhere' >> /home/foo/nodeSync.log 2>&1` to run this script every 10 mins.
```
Comment (morgwaiwetrust) [inline outOfSyncNoti.sh]: port param seems to be missing in the command
also let's move the command to the new line for better readability

File: outOfSyncNoti.sh [hunk 2]
```diff
@@ -1,14 +1,22 @@
...
+
+network=$1
+port=$2
```
Comment (morgwaiwetrust) [inline outOfSyncNoti.sh]: some comments explaining the params here would be nice:
- what is port? port of opsGenie API, port of geth API, port this script is listening on?
- what is treshold? number of blocks behind, unit of time it has been going on? what unit? minutes, seconds?

File: outOfSyncNoti.sh [hunk 3]
```diff
@@ -1,14 +1,22 @@
...
+genieKey=$3
+threshold=${4:-200}
 
 # Get current node block number
-curBlock=$(eval geth --exec "eth.blockNumber" attach http://localhost:8545)
+curBlock=$(eval geth --exec "eth.blockNumber" attach http://localhost:$port)
```
Comment (morgwaiwetrust) [inline outOfSyncNoti.sh]: why do you need `eval` here?

File: outOfSyncNoti.sh [hunk 4]
```diff
@@ -1,14 +1,22 @@
...
+if [ "$?" = "1" ]; then
```
Comment (morgwaiwetrust) [inline outOfSyncNoti.sh]: better use
`if [ $? -ne 0 ]; then`

File: outOfSyncNoti.sh [hunk 5]
```diff
@@ -1,14 +1,22 @@
...
+  curBlock=0
+fi
 
 # Get latest block number from etherscan
 # If on mainnet
-if [ "$1" == "mainnet" ]; then
+if [ "$network" == "mainnet" ]; then
```
Comment (morgwaiwetrust) [inline outOfSyncNoti.sh]: the below 2 commands differ only in etherscan subdomain name: let's introduce variable like `etherscanDomain` and assign it depending on network in this `if` and then perform the command.

File: outOfSyncNoti.sh [hunk 6]
```diff
@@ -21,12 +29,10 @@ echo "latest block number from etherscan: $ethBlock"
 echo "Our current block number: $curBlock"
```
Comment (morgwaiwetrust) [inline outOfSyncNoti.sh]: `echo "latest block number from localhost:${port} :"`

File: outOfSyncNoti.sh [hunk 7]
```diff
@@ -21,12 +29,10 @@ echo "latest block number from etherscan: $ethBlock"
...
 # Check if node is out of sync.
-# Will send notification when current block number is <numberofblocksbehind> blocks behind latest block on etherscan.
-genieKey=$2
-numberofblocksbehind=${3:-200}
-lowestBlock=$(($ethBlock-$numberofblocksbehind))
-
+# Will send notification when current block number is <threshold> blocks behind latest block on etherscan.
+lowestBlock=$(($ethBlock-$threshold))
 if [ $curBlock -lt $lowestBlock ];then
+  systemctl restart geth
```
Comment (morgwaiwetrust) [inline outOfSyncNoti.sh]: systemd service name should be a script para imo

File: outOfSyncNoti.sh [hunk 8]
```diff
@@ -21,12 +29,10 @@ echo "latest block number from etherscan: $ethBlock"
...
 
 # Check if node is out of sync.
-# Will send notification when current block number is <numberofblocksbehind> blocks behind latest block on etherscan.
-genieKey=$2
-numberofblocksbehind=${3:-200}
-lowestBlock=$(($ethBlock-$numberofblocksbehind))
-
+# Will send notification when current block number is <threshold> blocks behind latest block on etherscan.
+lowestBlock=$(($ethBlock-$threshold))
 if [ $curBlock -lt $lowestBlock ];then
```
Comment (morgwaiwetrust) [inline outOfSyncNoti.sh]: `if [ $(( ${ethBlock} - ${curBlock} )) -ge $threshold ]; then`
and remove misleading `lowestBlock`

Comment (kivutar) [reply outOfSyncNoti.sh]: If I do this, I need to change the default from `-200` to `200` right?

**Pull request merged**